conans/test/unittests/tools/cmake/test_cmake_test.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+
+from conan.tools.cmake import CMake
+from conan.tools.cmake.presets import write_cmake_presets
+from conans.client.conf import get_default_settings_yml
+from conans.model.conf import Conf
+from conans.model.settings import Settings
+from conans.test.utils.mocks import ConanFileMock
+from conans.test.utils.test_files import temp_folder
+
+
+@pytest.mark.parametrize("generator,target", [
+    ("NMake Makefiles", "test"),
+    ("Ninja Makefiles", "test"),
+    ("Ninja Multi-Config", "test"),
+    ("Unix Makefiles", "test"),
+    ("Visual Studio 14 2015", "RUN_TESTS"),
+    ("Xcode", "RUN_TESTS"),
+])
+def test_run_tests(generator, target):
+    """
+    Testing that the proper test target is picked for different generators, especially multi-config ones.
+    Issue related: https://github.com/conan-io/conan/issues/11405
+    """
+    settings = Settings.loads(get_default_settings_yml())
+    settings.os = "Windows"
+    settings.arch = "x86"
+    settings.build_type = "Release"
+    settings.compiler = "Visual Studio"
+    settings.compiler.runtime = "MDd"
+    settings.compiler.version = "14"
+
+    conanfile = ConanFileMock()
+    conanfile.conf = Conf()
+    conanfile.folders.generators = "."
+    conanfile.folders.set_base_generators(temp_folder())
+    conanfile.settings = settings
+
+    write_cmake_presets(conanfile, "toolchain", generator, {})
+    cmake = CMake(conanfile)
+    cmake.test()
+    assert "'--target' '%s'" % target in conanfile.command
```

Comment:
```suggestion
    if platform.system() == "Windows":
        assert "--target %s" % target in conanfile.command
    else:
        assert "'--target' '%s'" % target in conanfile.command
```

There is a small difference atm in the command in Windows, the args are not quoted.

Reply:
cc/ @lasote This is one of the things related to https://github.com/conan-io/conan/pull/11553, that seems a bit weird to me, that the commands quoting is different in different systems.

Reply:
Thank you for the suggestion. I added the `platform` import and made the search pattern conditional on the platform.